src/cminx/documentation_types.py

```diff
@@ -62,12 +62,25 @@ class DocumentationType(ABC):
     doc: str
     """The full doc comment, cleaned of # characters."""
 
-    @abstractmethod
     def process(self, writer: RSTWriter) -> None:
         """
         Processes the data stored in this documentation type,
         converting it to RST form by using the given RST writer.
 
+        :param writer: RSTWriter instance that the documentation
+                       will be added to.
+        """
+        if ':cminx: no-doc' in self.doc:
```

Comment:
I would recommend against checking for literal strings within the doccomment when possible, as it is error-prone and can lead to confusing errors. For example, if someone misspelled it as `:minx: no-doc` or forgot the space as in `:cminx:no-doc`, the doccomment would be output verbatim, and would likely result in errors during Sphinx generation.

CMinx already has support for parsing special-case doccomments that contain commands meant for CMinx itself, see the `@module` doccomment command. I would recommend implementing a similar mechanism for no-doc. This way, the no-doc doccomment is converted to its own token/parse rule, which can be handled by a no-op processor class. This would also eliminate the need for separate `process()` and `process_rst()` methods, since a no-doc will never be dispatched to any processor except the no-op. An example:

```cmake
#[[[ @no-doc #]]]
function(some_func)
    # Some commands
endfunction()

```

This would then be parsed into a `no-doc-command` rule, which consists of the no-doc doccomment block and the command invocation it is attached to.

Modifying the grammar a bit to disallow any `@`-commands in the doccomment header would make a misspelling of the `no-doc` command a parse error, which could point the developer directly to the line in the CMake source code that caused the error, instead of deferring it to Sphinx generation.

Reply:
Yep. Swallowing all `@` commands as part of the grammar is a good suggestion. After checking the `@module` command implementation I had a similar thought, noting it was part of the grammar. This looks like the best approach to maintaining consistency and providing useful error messages.

Reply:
It looks like the general consensus is that this PR should be implemented as `@no-doc` rather than as `:cminx: no-doc`. Is that correct? If so, @JackNoordhuis does it make sense to keep this PR open? If we decide to close it, I'll create an issue and try to find some time to work on adjusting the grammar.

Reply:
@ryanmrichard Yes, this PR is obsolete. I have a [branch on my fork](https://github.com/CMakePP/CMinx/compare/master...JackNoordhuis:CMinx:no-doc-docstring) that has passing tests for `@` doc-string support. I've been busy this week and got sidetracked writing benchmarks for my changes so I haven't opened a new PR yet. My grammar changes just allow the AST visitor to process doc-comments with lines containing the `@` commands separately (refactors the old module nodes to generic doc command nodes.)  I spent a few hours trying to get the ANTLR grammar to parse the doc-comments (clean the lines, find `@` commands, etc.) but settled on parsing the command names and parameters by hand in Python. There are some more in-depth changes to the processing/handling of the commands themselves (escaped spaces in parameters, escaping newlines to continue parameters, etc.) that I've made with plugin/extension support in mind (future support for different output formats?)  mostly inspired by doxygen-style commands. I should have time to clean up the code and benchmark suite to open separate PR's sometime this weekend.

Reply:
@JackNoordhuis That sounds amazing. I really appreciate the interest and willingness to help add a new feature!